src/main/java/com/sonyericsson/hudson/plugins/gerrit/trigger/playback/GerritMissedEventsPlaybackManager.java

```diff
@@ -0,0 +1,438 @@
+
+package com.sonyericsson.hudson.plugins.gerrit.trigger.playback;
+
+import com.sonyericsson.hudson.plugins.gerrit.trigger.GerritServer;
+import com.sonyericsson.hudson.plugins.gerrit.trigger.PluginImpl;
+import com.sonyericsson.hudson.plugins.gerrit.trigger.config.IGerritHudsonTriggerConfig;
+import com.sonyericsson.hudson.plugins.gerrit.trigger.utils.GerritPluginChecker;
+import com.sonymobile.tools.gerrit.gerritevents.ConnectionListener;
+import com.sonymobile.tools.gerrit.gerritevents.GerritEventListener;
+import com.sonymobile.tools.gerrit.gerritevents.GerritJsonEventFactory;
+import com.sonymobile.tools.gerrit.gerritevents.dto.GerritEvent;
+import com.sonymobile.tools.gerrit.gerritevents.dto.attr.Provider;
+import com.sonymobile.tools.gerrit.gerritevents.dto.events.ChangeBasedEvent;
+import com.sonymobile.tools.gerrit.gerritevents.dto.events.GerritTriggeredEvent;
+import hudson.Util;
+import hudson.XmlFile;
+import net.sf.json.JSONObject;
+import org.apache.commons.io.IOUtils;
+import org.apache.http.HttpHost;
+import org.apache.http.HttpResponse;
+import org.apache.http.auth.AuthScope;
+import org.apache.http.client.methods.HttpGet;
+import org.apache.http.conn.params.ConnRoutePNames;
+import org.apache.http.impl.client.DefaultHttpClient;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.UnsupportedEncodingException;
+import java.net.HttpURLConnection;
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.net.URLEncoder;
+import java.text.SimpleDateFormat;
+import java.util.Date;
+import java.util.Map;
+import java.util.Scanner;
+import java.util.Vector;
+import java.util.concurrent.ConcurrentHashMap;
+
+
+/**
+ * The GerritMissedEventsPlaybackManager is responsible for recording a last-alive timestamp
+ * for each server connection. The motivation is that we want to be able to know when we last
+ * received an event. This will help us determine upon connection startup, if we have missed
+ * some events while the connection was down.
+ *
+ * Once the server is re-connected, the missed event will be played back as if they had been
+ * received orginally.
+ *
+ * @author scott.hebert@ericsson.com
+ */
+public class GerritMissedEventsPlaybackManager implements ConnectionListener, GerritEventListener {
+
+    private static final Logger logger = LoggerFactory.getLogger(GerritMissedEventsPlaybackManager.class);
+    static final String EVENTS_LOG_PLUGIN_NAME = "events-log";
+    private static final String EVENTS_LOG_PLUGIN_URL = "a/plugins/" + EVENTS_LOG_PLUGIN_NAME + "/events/";
+
+    private String serverName;
+    /**
+     * Map that contains timestamps.
+     */
+    protected Map<String, EventTimeSlice> serverTimestamps = new ConcurrentHashMap<String, EventTimeSlice>();
+    /**
+     * Vector that contains received Gerrit Events.
+     */
+    protected Vector<ChangeBasedEvent> receivedEventCache = new Vector<ChangeBasedEvent>();
```

Comment:
I would prefer you use ```List l = Collections.synchronizedList(new SomeListImplementation())``` as it doesn't expose any specific implementation as well as clearly state synchronization intent instead of giving Java 1.1 flashbacks.